### PR TITLE
Give ImagePolicy objects a readiness condition

### LINKED ISSUE #66: ImagePolicy readiness
While writing the CLI subcommands for image policy, I realised that there's not a reliable indication of readiness for an image policy. Although they are just a calculation based on the image data, there's still things that can go wrong e.g., the referenced image repository does not exist, or is not ready itself.

Therefore,

 - [x] add ready condition
 - [x] add condition metric

(any other GOTK controller has examples of how to do these)

### PR DESCRIPTION
**Pull request by squaremo**
This adds the conventional Conditions field to the ImagePolicy status, and sets the status in the controller as well as reporting readiness metrics.

Fixes #66.

### CODE REVIEW

**Comment by squaremo** [comment]
One problem still extant is that the controller won't set a status if there's no tags that match. I'll fix that here too.

File: controllers/imagepolicy_controller.go [hunk 1]
```diff
@@ -83,6 +86,7 @@ func (r *ImagePolicyReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 			return ctrl.Result{}, err
 		}
 		defer r.MetricsRecorder.RecordDuration(*objRef, reconcileStart)
+		defer r.recordReadinessMetric(&pol)
```
[the remark this answers is not in the document]
Comment (squaremo) [reply controllers/imagepolicy_controller.go]: I've changed it to be in line with the other controller here.

**Review by relu — APPROVED** [review]
LGTM :sunglasses:

**Comment by squaremo** [comment]
Thanks for reviewing @relu 🥭

**Pull request merged**